### ISSUE TITLE
Enhance User Experience with Website Translation Feature on GitHub

### ISSUE THREAD

**Issue by XCOD33** [raymondmars/chatgpt-chrome-translate-plugin#25]
Please add a feature for translating the entire website. Having to block each piece of text to be translated is very cumbersome. You might find a similar feature in the Sider.AI extension.

**Comment by raymondmars** [comment]
There are a couple of extensions that have implemented this function which can translate the whole text of a web page before I made this extension. I saw it. but I disliked that idea, so I created this extension.
In my opinion, the idea of translating the whole page has these problems:
1. It many times can not be controlled by the user freely and will let the bot translate contents that you do not want to be translated. The page will be changed in the future, If you want to use one rule to apply to all of the pages that is impossible. It leads the code to become more complex and you will continue to add new patch codes while the pages change.
2. Due to the cost of the ChatGPT API, translating the whole page content is a costly task. some pages are small, others could contain huge content and the user can't estimate the cost before they enjoy the whole page translating. If someday, we have a free bot API, then It could be a good time to add this function.
Another reason that led me to not add this function is that I prefer to learn the language while I'm reading. I guess some people have the same idea as me and wouldn't like all of the content of the page to be translated into the target language. 
Thank you for your suggestions and welcome to give me more advice.

**Comment by XCOD33** [comment]
That's true what you said. but adding that feature can also give users options. blocking every sentence that wants to be translated is very painful, especially if the user is reading a long article.

but again, this extension is already good, the option to add such features is back to you. so far I appreciate you for making this extension.

(other extensions that can translate the whole web are mostly paid, there are free ones but who needs other options like summarisers when users just need a translator extension)

**Comment by raymondmars** [comment]
@XCOD33 I have an idea that we can add this function to the extension but only optimize it for some popular websites such as the Economist, Reddit, Bloomberg, New York Times, and so on. The user can send their favorite sites to me and I will evaluate these sites to decide if they could be added in the end. From my perspective, it will ensure the user can get a good experience when they use this extension. I will write a rule engine to implement this function later. Thank you for your fantastic insights again.
